src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_static.c

```diff
@@ -44,19 +45,19 @@ EMSCRIPTEN_KEEPALIVE int32_t mono_wasm_load_icu_data(void * pData)
         //// see https://github.com/unicode-org/icu/blob/master/docs/userguide/icu_data/tracing.md
         // utrace_setFunctions(0, 0, 0, icu_trace_data);
         // utrace_setLevel(UTRACE_VERBOSE);
+        isDataSet = 1;
         return 1;
     }
 }
 #endif
 
 int32_t GlobalizationNative_LoadICU(void)
 {
-    const char* icudir = getenv("DOTNET_ICU_DIR");
-    if (icudir)
-        u_setDataDirectory(icudir);
-    else
-        ; // default ICU search path behavior will be used, see http://userguide.icu-project.org/icudata
-
+    if (!isDataSet) {
+        // don't try to locate icudt.dat automatically if mono_wasm_load_icu_data wasn't called
+        // and fallback to invariant mode
+        return 0;
```

Comment:
This won't cause to fallback to Invariant mode... this would fail fast because in the GlobalozationMode.Unix implementation we failfast if LoadICU returns false.

Reply:
Yeah

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs#L23

Reply:
> This won't cause to fallback to Invariant mode... this would fail fast because in the GlobalozationMode.Unix implementation we failfast if LoadICU returns false.

ah, right, windows impl confused me where to goes back to NLS if LoadICU fails.